src/components/GuestBook/GuestBook.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react'
+import { useNotes } from '../../context/NoteContext'
+import { useUser } from '../../context/UserContext'
+
+
+export default function GuestBook() {
+    // const [name, setName] = useState('')
+    const [inputNote, setInputNote] = useState('')
+    //now we we need to access the Note state and User state 
+    const { notes, setNotes } = useNotes()
+    const { user, setUser} = useUser()
+
+    function NameUpdate(){
+        //I need to be able to update the name of the user here...
+        //if it is not a note from the user, return
+        if (!inputNote) return
+         
+        
+        //then...with setNotes, I will need to set the message of that user
+        //we want to spread the notes so that it will render all of the notes we have
+        //we then want to be able to set the user name and their message
+        //this is the larger notes that is being kept globally
+        setNotes([...notes, {name: user, message: inputNote}])
+        //then i will need to setGuessNote...to maybe an empty string?
+        //after user hits the button to add the note, this will render an empty string to clear it out
+        setInputNote('')
+    }
+
+    const handleSubmit = (e) =>{
+        e.preventDefault()
+        NameUpdate()
```

Comment:
this feels a bit clunky now that the name is coming from the `useUser` hook; I think that moving lines 23 & 26 down here and doing away with `NameUpdate` would contribute to the readability here.